waltz-server/src/main/java/com/wepay/waltz/server/internal/WaltzServerHandler.java

```diff
@@ -112,6 +117,12 @@ protected void process(Message msg) throws Exception {
                     storageConnectivityMap), true);
                 break;
 
+            case MessageType.SERVER_PARTITIONS_ASSIGNMENT_REQUEST:
+                List<Integer> partitionsAssigned = new ArrayList<>(partitions.keySet());
```

Comment:
shouldn't `partitions` be accessed inside the `synchronized (partitions) {}` ??

Reply:
Yeah, good catch!

Reply:
fixed. Thanks!